include/common/ast/module.h

```diff
@@ -57,19 +59,66 @@ class Module : public Base {
   CodeSection *getCodeSection() const { return CodeSec.get(); }
   DataSection *getDataSection() const { return DataSec.get(); }
 
-  using TrapProxy = void (*)(Interpreter::Interpreter *, uint32_t);
-  using CallProxy = void (*)(Interpreter::Interpreter *, const uint32_t,
-                             const ValVariant *, ValVariant *);
-  using MemGrowProxy = uint32_t (*)(Interpreter::Interpreter *, const uint32_t);
-  using MemSizeProxy = uint32_t (*)(Interpreter::Interpreter *);
-  using Ctor = void (*)(TrapProxy, CallProxy, MemGrowProxy, MemSizeProxy);
+  using TrapCodeProxy = uint32_t *;
+  using CallProxy = void (*)(const uint32_t FuncIdx, const ValVariant *Args,
+                             ValVariant *Rets);
+  using MemGrowProxy = uint32_t (*)(const uint32_t Diff);
 
-  Ctor getCtor() const { return CtorFunc; }
-  void setCtor(Ctor F) { CtorFunc = F; }
+  void setTrapCodeProxySymbol(TrapCodeProxy *Symbol) {
+    TrapCodeProxySymbol = Symbol;
+  }
+  void setCallProxySymbol(CallProxy *Symbol) { CallProxySymbol = Symbol; }
+  void setMemGrowProxySymbol(MemGrowProxy *Symbol) {
+    MemGrowProxySymbol = Symbol;
+  }
+  void setTrapCodeProxy(TrapCodeProxy Pointer) const {
+    if (TrapCodeProxySymbol)
+      *TrapCodeProxySymbol = Pointer;
+  }
+  void setCallProxy(CallProxy Pointer) const {
+    if (CallProxySymbol)
+      *CallProxySymbol = Pointer;
+  }
+  void setMemGrowProxy(MemGrowProxy Pointer) const {
+    if (MemGrowProxySymbol)
+      *MemGrowProxySymbol = Pointer;
+  }
 
   /// The node type should be ASTNodeAttr::Module.
   const ASTNodeAttr NodeAttr = ASTNodeAttr::Module;
 
+  static std::string toExportName(std::string_view Name) {
+    using namespace std::literals::string_view_literals;
+    const auto NPos = std::string_view::npos;
+    const auto Esc = "$#@\0"sv;
+    std::string Result(1, '$');
+    size_t Cursor = 0;
+    for (size_t Pos = Name.find_first_of(Esc); Pos != NPos;
+         Cursor = Pos + 1, Pos = Name.find_first_of(Esc, Cursor)) {
+      Result += Name.substr(Cursor, Pos - Cursor);
+      switch (Name[Pos]) {
+      case '$':
```

Comment:
Maybe we should have a table or document to record the mapping relations?

Reply:
Done.